documentation/library_guide/dynamic_selection_api/fixed_resource_policy.rst

```diff
@@ -0,0 +1,242 @@
+  
+  }
+  
+This policy can be used with all of the dynamic selection functions, such as ``select``, ``submit``,
+and ``submit_and_wait``. It also can used with ``policy_traits``.
+
+Example
+-------
+
+In the following example, a ``fixed_resource_policy`` is used when the code is
+compiled so that it selects a specific device.  When ``USE_CPU`` is defined at 
+compile-time, this example will always use the CPU queue. When ``USE_GPU`` is defined 
+at compile-time, it will always use the GPU queue. Otherwise, it uses an 
+``auto_tune_policy`` to dynamically select between these two queues. Such a scenario 
+could be used for debugging or simply to maintain the dynamic selection code even if 
+the best device to use is know for some subset of platforms.  
+
+.. code:: cpp
+
+  #include <oneapi/dpl/dynamic_selection>
+  #include <sycl/sycl.hpp>
+  #include <iostream>
+
+  namespace ex = oneapi::dpl::experimental;
+
+  int main() {
+    std::vector<sycl::queue> r { sycl::queue{sycl::cpu_selector_v},
+                                 sycl::queue{sycl::gpu_selector_v} };
+
+    const std::size_t N = 10000;
+    std::vector<float> av(N, 0.0);
+    std::vector<float> bv(N, 0.0);
+    std::vector<float> cv(N, 0.0);
+    for (int i = 0; i < N; ++i) {
+      av[i] = bv[i] = i;
+    }
+
+  #if USE_CPU
+    ex::fixed_resource_policy p{r};    // (1) uses index 0 of r, the cpu
+  #elif USE_GPU
+    ex::fixed_resource_policy p{r, 1}; // (2) uses index 1 of r, the gpu
+  #else 
+    ex::auto_tune_policy p{r};
+  #endif
+
+    {
+      sycl::buffer<float> ab(av);
+      sycl::buffer<float> bb(bv);
+      sycl::buffer<float> cb(cv);
+
+
+      for (int i = 0; i < 6; ++i) {
+        ex::submit_and_wait(p, [&](sycl::queue q) { // (3)
+          // (4)
+          std::cout << (q.get_device().is_cpu() ? "using cpu\n" : "using gpu\n");
+          return q.submit([&](sycl::handler &h) {   // (5)
+            sycl::accessor aa(ab, h, sycl::read_only);
+            sycl::accessor ba(bb, h, sycl::read_only);
+            sycl::accessor ca(cb, h, sycl::read_write);
+            h.parallel_for(N, [=](auto i) { ca[i] = aa[i] + ba[i]; }); 
+          });
+        }); 
+      };  
+    }
+
+    for (int i = 0; i < N; ++i) {
+      if (cv[i] != 2*i) {
+         std::cout << "ERROR!\n";
+      }
+    }
+    std::cout << "Done.\n";
+  }
+  
+The key points in this example are:
+
+#. If ``USE_CPU`` is defined, a ``fixed_resouce_policy`` is constructed that targets the CPU.
+#. If ``USE_GPU`` is defined, a ``fixed_resouce_policy`` is constructed that targets the GPU.
+#. ``submit_and_wait`` is invoked with the policy as the first argument. The user-provided function will be passed the selected queue.
+#. For clarity when run, the type of device is displayed.
+#. The queue is used in function to perform an asynchronous offload. The SYCL event returned from the call to ``submit`` is returned. Returning an event is required for functions passed to ``submit`` and ``submit_and_wait``.

```

Comment:
No it is still required that the user-provided function returns an event. This is because a policy only needs to support select and submit. The default implementation therefore uses select, submit and finally waits on the event returned from submit to implement submit_and_wait. So, the user-provided function must return an event so that the default implementation works.

Reply:
But the submit_and_wait function itself now returns void.